### DISCUSSION TITLE
Difference of useBatchFinishListener and useBatchFinalizeListener

### DISCUSSION THREAD

**Discussion by smartlightning** [rpldy/react-uploady#406] [Q&A] [answered]
From the documentation I don't understand what the difference between useBatchFinishListener and useBatchFinalizeListener is. Can someone please explain it to me?

**Comment by yoavniran** [comment, marked as answer]
@smartlightning Finish is called when the uploading is done. Finalize is always called; no matter how uploading was terminated (success, error, abort, etc)
I will check the docs and see if I can make it clearer
